util/EIP-152/src/avx.rs

```diff
@@ -0,0 +1,405 @@
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g2(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                diagonalize(&mut a, &mut b, &mut c, &mut d);
+                t0 = _mm256_blend_epi32(m2, m1, 0x33);
+                t1 = _mm256_alignr_epi8(m4, m7, 8);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g1(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                t0 = _mm256_unpacklo_epi64(m5, m0);
+                t1 = _mm256_unpacklo_epi64(m2, m3);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g2(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                undiagonalize(&mut a, &mut b, &mut c, &mut d);
+            },
+            8 => {
+                // round 9
+                t0 = _mm256_unpacklo_epi64(m3, m7);
+                t1 = _mm256_alignr_epi8(m0, m5, 8);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g1(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                t0 = _mm256_unpackhi_epi64(m7, m4);
+                t1 = _mm256_alignr_epi8(m4, m1, 8);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g2(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                diagonalize(&mut a, &mut b, &mut c, &mut d);
+                t0 = _mm256_unpacklo_epi64(m5, m6);
+                t1 = _mm256_unpackhi_epi64(m6, m0);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g1(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                t0 = _mm256_alignr_epi8(m1, m2, 8);
+                t1 = _mm256_alignr_epi8(m2, m3, 8);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g2(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                undiagonalize(&mut a, &mut b, &mut c, &mut d);
+            },
+            9 => {
+                // round 10
+                t0 = _mm256_unpacklo_epi64(m5, m4);
+                t1 = _mm256_unpackhi_epi64(m3, m0);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g1(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                t0 = _mm256_unpacklo_epi64(m1, m2);
+                t1 = _mm256_blend_epi32(m2, m3, 0x33);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g2(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                diagonalize(&mut a, &mut b, &mut c, &mut d);
+                t0 = _mm256_unpackhi_epi64(m6, m7);
+                t1 = _mm256_unpackhi_epi64(m4, m1);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g1(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                t0 = _mm256_blend_epi32(m5, m0, 0x33);
+                t1 = _mm256_unpacklo_epi64(m7, m6);
+                b0 = _mm256_blend_epi32(t0, t1, 0xF0);
+                g2(&mut a, &mut b, &mut c, &mut d, &mut b0);
+                undiagonalize(&mut a, &mut b, &mut c, &mut d);
+            },
+            _ => unreachable!("numbers 0-10 covered; qed")
```

Comment:
when we have benchmarks, would be nice to see if removing the unreachable statement by replacing `9 =>` with `_=>` makes a difference